bindings/python/setup.py

```diff
@@ -15,16 +18,26 @@ def get_pymongoarrow_version():
 
 
 def get_extension_modules():
-    modules = cythonize(['pymongoarrow/*.pyx',
-                         'pymongoarrow/libbson/*.pyx'])
-    for module in modules:
+    arrow_modules = cythonize(['pymongoarrow/*.pyx'])
+    libbson_modules = cythonize(['pymongoarrow/libbson/*.pyx'])
+
+    for module in libbson_modules:
         module.libraries.append('bson-1.0')
-    return modules
+
+    for module in arrow_modules:
+        module.include_dirs.append(np.get_include())
+        module.include_dirs.append(pa.get_include())
+        module.libraries.extend(pa.get_libraries())
+        module.library_dirs.extend(pa.get_library_dirs())
+        if os.name == 'posix':
+            module.extra_compile_args.append('-std=c++11')
```

Comment:
Could you add a comment linking to https://arrow.apache.org/docs/python/extending.html#example so we know where this pattern comes from?

Reply:
Done.